server/src/main/scala/viper/gobraserver/GobraServer.scala

```diff
@@ -51,50 +53,61 @@ object GobraServer extends GobraFrontend {
     _server.start()
     VerifierState.flushCachedDiagnostics()
   }
-  
 
-  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]) {
+  def restart() {
+    _server.stop()
+    ViperBackends.ViperServerBackend.resetServer()
+    init(_options)
```

Comment:
There are a few things here that seem odd to me.
1. I assume _server.stop() and _.resetServer() are blocking, right?
2. First you reset the server and then within init(...) the server is reassigned. Is resetServer more like stopping the server, or why do you have to reset it before you just reassign it?
3. If we consider resource ownership, then the call to init with _option is problematic. The method requiers read permissions on the argument and write permissions on _option. I am aware that this is not a problem in this particular case, but I would feel more comfortable if you could copy the argument before the call.

Reply:
1. They are not really blocking as `_server.stop()` sends some exit message via Akka actors that everything needs to shutdown. I tried to adapt it to return a future but I don't know whether it's possible to wait on completion processing the exit message. `_.resetServer()` only sets a local variable to `null`.
2. A reset is necessary as `init(...)` requires that no server is already set
3. I can do that